src/pysdccc/_cli.py

```diff
@@ -0,0 +1,135 @@
+import contextlib
+import io
+import pathlib
+import subprocess
+import sys
+import tempfile
+import zipfile
+
+import httpx
+
+from pysdccc import _runner
+
+try:
+    import click
+except ImportError as import_error:
+    msg = "Cli not installed. Please install using 'pip install pysdccc[cli]."
+    raise ImportError(msg) from import_error
+
+
+class UrlType(click.ParamType):
+    name = 'url'
+
+    def convert(self, value: str, param: click.Parameter | None, ctx: click.Context | None) -> httpx.URL:
+        try:
+            return httpx.URL(value)
+        except Exception as e:  # noqa: BLE001
+            self.fail(f'{value!r} is not a valid url: {e}', param, ctx)
+
+
+URL = UrlType()
+
+
+class ProxyType(click.ParamType):
+    name = 'proxy'
+
+    def convert(self, value: str, param: click.Parameter | None, ctx: click.Context | None) -> httpx.Proxy:
+        try:
+            return httpx.Proxy(value)
+        except Exception as e:  # noqa: BLE001
+            self.fail(f'{value!r} is not a valid proxy: {e}', param, ctx)
+
+
+PROXY = ProxyType()
+
+
+def _download_to_stream(
```

Comment:
Why doesn't this use the download_to_stream function defined in _download? It seems like this should only exist once and progress can be controlled via a callback.

Reply:
I reduced amount of code duplication

Reply:
@ldeichmann ask me to take over, solved for me